src/identity/IdentityProviderHttpHandler.ts

```diff
@@ -162,39 +187,79 @@ export class IdentityProviderHttpHandler extends HttpHandler {
    * Handles the behaviour of an InteractionRoute.
    * Will error if the route does not support the given request.
    *
-   * GET requests go to the templateHandler, POST requests to the specific InteractionHandler of the route.
+   * GET requests return a default response result,
+   * POST requests to the specific InteractionHandler of the route.
    */
-  private async resolveRoute(request: HttpRequest, response: HttpResponse, route: InteractionRoute,
-    oidcInteraction?: Interaction): Promise<void> {
-    if (request.method === 'GET') {
+  private async resolveRoute(operation: Operation, route: InteractionRoute, oidcInteraction?: Interaction):
+  Promise<{ result: InteractionHandlerResult; templateFiles: Record<string, string> }> {
+    if (operation.method === 'GET') {
       // .ejs templates errors on undefined variables
-      return await this.handleTemplateResponse(response, route.viewTemplate, { errorMessage: '', prefilled: {}});
+      return {
+        result: { type: 'response', details: { errorMessage: '', prefilled: {}}},
```

Comment:
Weird to have those empty things there. Omit?

Reply:
Union typing on result?

Reply:
They're not here because of typing issues, they're here because the EJS templates crash if the variables are not defined. `if (errorMessage)` errors if `errorMessage` was not passed as a parameter to the template. Didn't look into this fully, but I'm guessing there are probably ways to get around this by editting the EJS templates.

Reply:
Yeah, with `if (typeof errorMessage !== 'undefined')` is pretty inelegant.

But can whatever component that calls the template not fill those out?

Reply:
We don't need it for all templates though, and once out of this function the next step is the TemplateEngine itself. Will probably still look into a way to get this out of the JSON response body at some point but that is less urgent.